geniza/corpus/management/commands/import_data.py

```diff
@@ -336,3 +372,153 @@ def log_creation(self, *objects):
                 object_repr=str(obj),
                 change_message=self.logentry_message,
                 action_flag=ADDITION)
+
+    def get_user(self, name, pgpid=None):
+        """Find a user account based on a provided name, using a simple cache.
+
+        If not found, tries to use first/last initials for lookup. If all else
+        fails, use the generic team account (TEAM_USERNAME).
+        """
+
+        # check the cache first
+        user = self.user_lookup.get(name)
+        if user:
+            logging.debug(
+                f"Using cached user {user} for {name} on PGPID {pgpid}")
+            return user
+
+        # person with given name(s) and last name – case-insensitive lookup
+        if " " in name:
+            given_names, last_name = [sname.strip(punctuation) for sname in
+                                      name.rsplit(" ", 1)]
+            try:
+                user = User.objects.get(first_name__iexact=given_names,
+                                        last_name__iexact=last_name)
+            except User.DoesNotExist:
+                pass
+
+        # initials; use first & last to do lookup
+        elif name:
+            first_i, last_i = name[0], name[-1]
+            try:
+                user = User.objects.get(first_name__startswith=first_i,
+                                        last_name__startswith=last_i)
+            except User.DoesNotExist:
+                pass
+
+        # if we didn't get anyone through either method, warn and use team user
+        if not user:
+            logging.warning(
+                f"Couldn't find user {name} on PGPID {pgpid}; using {self.team_user}")
+            return self.team_user
```

Comment:
It didn't seem simple to derive that from the spreadsheet they gave me – a lot of the entries were things I didn't need, and others I would have had to transform to make them usable. Overall there are only two ways to find a person (name or initials), so I thought this was the most straightforward way.

Reply:
yeah, I didn't expect you to generate it from the table they gave you; thought maybe you could generate from the fixture list once you had it, since you know the variants that occur